src/test/tests_common.h

```diff
@@ -62,12 +62,25 @@ struct test_options {
 	int	self_rank;
 	int	mypid;
 	int	is_server;
+	int	num_attach_retries;
+	int	do_assert;
 };
 
 struct test_options opts;
 
 int g_shutdown;
 
+void
+tc_test_init(int rank, int pid, int is_server, int num_attach_retries,
```

Comment:
I've posted comments on this usage throughout, instead i ll just summarize here:
1. pid seems to be not needed to be passed; i cant think of a case when something other than getpid() would be passed.
2. for rank d_rank_t type should be used
3. for is_server/do_assert its better to use bool type and pass true/false values
4. in my view do_assert better to rename to assert_on_error
5. It's more of a preference to have complex args first followed by true/false flags later so something like this instead:
tc_test_init(d_rank_t rank, int num_atach_retries, bool is_server, bool assert_on_err, bool some_other_flag, etc...)